source/CollisionSet.cpp

```diff
@@ -314,26 +311,26 @@ const vector<Collision> &CollisionSet::Line(const Point &from, const Point &to,
 
 
 // Get all objects within the given range of the given point.
-const vector<Body *> &CollisionSet::Circle(const Point &center, double radius) const
+vector<Body *> &CollisionSet::Circle(const Point &center, double radius, vector<Body *> &result) const
 {
-	return Ring(center, 0., radius);
+	return Ring(center, 0., radius, result);
 }
 
 
 
 // Get all objects touching a ring with a given inner and outer range
 // centered at the given point.
-const vector<Body *> &CollisionSet::Ring(const Point &center, double inner, double outer) const
+vector<Body *> &CollisionSet::Ring(const Point &center, double inner, double outer, vector<Body *> &circleResult) const
```

Comment:
The [review-feedback](https://github.com/endless-sky/endless-sky/pull/10217/files#r1643239850) from @RisingLeaf still seems relevant here.